README.md

```diff
@@ -12,22 +12,65 @@ by the generated snapshot flakes. However, the `flake.lock` in this repo
 is critically important, as it controls which versions of our upstream
 dependencies (nixpkgs, nix-ros-overlay) we are pinned to.
 
-## Usage
+## Nix Installation
+
+You can get Nix by the upstream-suggested `curl` install, by installing the
+apt package on an OS new enough to support it, or by using the the container.
 
-Add the repositories to your nix flake registry (so flakes need to be enabled)
-with:
+To use [the upstream-suggested installation][nix-install], run:
+```
+sh <(curl -L https://nixos.org/nix/install) --daemon
+```
 
+Or, if you're on Ubuntu Jammy+, you can bootstrap Nix directly
+from a distro package:
+```
+sudo apt install -y nix-setup-systemd
+
+# Optional but recommended; allows nix usage without root. Must log out
+# and in following this invocation:
+sudo usermod -a -G nix-users $(whoami)
+```
+
+As a final alternative, you can pull and use the
+[Nix container][nix-container] in docker or podman.
+
+[nix-install]: https://nixos.org/download.html
+[nix-container]: https://hub.docker.com/r/nixos/nix/
+
+## Nix Configuration
+
+Our work heavily leverages the upcoming [Flakes][nix-flakes] feature, so this
+must be manually enabled in your Nix environment:
+```
+# Or to /etc/nix/nix.conf if you're running Nix as root.
+echo "experimental-features = nix-command flakes" >> ~/.config/nix/nix.conf
+```
+
+Add the repositories to your nix flake registry with:
 ```
 nix registry add ros github:clearpathrobotics/nix-ros
 nix registry add ros-base github:clearpathrobotics/nix-ros-base
 ```
 
-To avoid rebuilding, one can optionally use the public binary cache for
-[nix-ros][nix-ros-cachix] on cachix.
+And finally, we suggest [setting up Cachix][nix-ros-cachix] so that you can
+pull pre-built binaries rather than building everything locally:
+```
+# Enter a temporary shell where cachix is available.
+nix shell github:cachix/cachix
+cachix use nix-ros
+exit
```

Comment:
Oh, this is nice, that way you don't end up with this persistently living in your store.

Reply:
I think you could even just do a one-off `nix run` but it's nice to demonstrate nix's ephemeral subshell workflow here.